### DISCUSSION TITLE
Running into valid download error

### DISCUSSION THREAD

**Discussion by redricka** [grahampugh/erase-install#334] [General]
Hi, 
Running this through Jamf on intel chip Macbooks, it was working a few days ago for a few machines but now peramently failing. 
I have thrown the eraseinstall pkg up in jamf repo to distribute to the machines, and in files and processors have the command 
'sudo /Library/Management/erase-install/erase-install.sh --force-curl --os=12.6 --overwrite --erase'

Returns the following. 

[erase-install] v27.2 script execution started: Wed Jan 18 14:48:07 AEDT 2023
   [erase-install] Caffeinating this script (pid=7442)
   [check_free_space] OK - 208 GB free/purgeable disk space detected
   [erase-install] Looking for existing installer app or pkg
   [find_existing_installer] No valid installer found.
   [overwrite_existing_installer] Deleting existing cached installer(s).
   [erase-install] Running on architecture i386
   [erase-install] Opening jamfHelper download message (language=en)
   [check_installinstallmacos] force-curl selected so installinstallmacos.py will be overwritten
   [download_installinstallmacos] Downloading installinstallmacos.py...
   [check_installinstallmacos] installinstallmacos.py is in /Library/Management/erase-install
   [check_python] Relocatable Python is installed in /Library/Management/erase-install
   [run_installinstallmacos] Checking that selected OS 12.6 is available

   [run_installinstallmacos] This command is now being run:

   installinstallmacos.py --workdir /Library/Management/erase-install --ignore-cache --raw --os 12.6 --validate

installinstallmacos.py - get macOS installers from the Apple software catalog

This Mac:
Model Identifier : MacBookAir9,1
 #  ProductID       Version    Build    Post Date   Title                          
 1  061-26578       10.14.5    18F2059  2019-10-14  macOS Mojave                   
 2  061-26589       10.14.6    18G103   2019-10-14  macOS Mojave                   
 3  041-91758       10.13.6    17G66    2019-10-19  macOS High Sierra              
 4  041-88800       10.14.4    18E2034  2019-10-23  macOS Mojave                   
 5  041-90855       10.13.5    17F66a   2019-10-23  Install macOS High Sierra Beta 
 6  061-86291       10.15.3    19D2064  2020-03-23  macOS Catalina                 
 7  001-04366       10.15.4    19E2269  2020-05-04  macOS Catalina                 
 8  001-15219       10.15.5    19F2200  2020-06-15  macOS Catalina                 
 9  001-36735       10.15.6    19G2006  2020-08-06  macOS Catalina                 
10  001-36801       10.15.6    19G2021  2020-08-12  macOS Catalina                 
11  001-51042       10.15.7    19H2     2020-09-24  macOS Catalina                 
12  001-57224       10.15.7    19H4     2020-10-27  macOS Catalina                 
13  001-68446       10.15.7    19H15    2020-11-11  macOS Catalina                 
14  071-78704       11.5.2     20G95    2021-08-18  macOS Big Sur                  
15  002-23589       11.6.1     20G224   2021-12-01  macOS Big Sur                  
16  002-42341       11.6.2     20G314   2022-01-14  macOS Big Sur                  
17  002-57023       11.6.3     20G415   2022-01-26  macOS Big Sur                  
18  002-65695       11.6.4     20G417   2022-02-17  macOS Big Sur                  
19  002-77154       11.6.5     20G527   2022-04-11  macOS Big Sur                  
20  012-08272       11.6.6     20G624   2022-05-24  macOS Big Sur                  
21  012-40494       12.6       21G115   2022-09-20  macOS Monterey                 
22  012-38280       11.7       20G817   2022-09-20  macOS Big Sur                  
23  012-92138       13.0       22A380   2022-10-24  macOS Ventura                  
24  012-90254       12.6.1     21G217   2022-10-24  macOS Monterey                 
25  012-90253       11.7.1     20G918   2022-10-24  macOS Big Sur                  
26  012-93766       13.0.1     22A400   2022-11-09  macOS Ventura                  
27  012-60271       13.1       22C65    2023-01-11  macOS Ventura                  
28  032-12843       12.6.2     21G320   2023-01-11  macOS Monterey                 
29  032-12833       11.7.2     20G1020  2023-01-11  macOS Big Sur                  

A valid installer for OS 12.6 is not available for this computer. Run again without --os argument to select a valid build to download.

   [run_installinstallmacos] No disk image found. I guess nothing got downloaded.

   [erase-install] attempting to terminate the 'caffeinate' process - Termination message indicates success
/Library/Management/erase-install/erase-install.sh: line 1046:  7459 Terminated: 15          /usr/bin/caffeinate -dimsu -w $$

   [erase-install] attempting to terminate the 'jamfHelper' process - Termination message indicates success
/Library/Management/erase-install/erase-install.sh: line 1046:  7483 Terminated: 15          "$jamfHelper" -windowType hud -windowPosition ul -title "${!dialog_dl_title}" -alignHeading center -alignDescription left -description "${!dialog_dl_desc}" -lockHUD -icon "$dialog_dl_icon" -iconSize 100

**Comment by grahampugh** [comment]
--os=12.6 is not a valid option. Choose --os=12 or --version=12.6. 

By the way, the current version is 12.6.2. --os=12 will provide you with 12.6.2. 

Also you don't need sudo in your run command. Jamf runs everything as root anyway.
